docs/adding-an-endpoint.adoc

```diff
@@ -0,0 +1,59 @@
+:showtitle:
+:numbered:
+:toc: left
+
+= Adding an Endpoint to Nexus
+
+Adding new endpoints cuts across many concepts in Nexus, from the HTTP
+interface, authorization, the database, and several tests. This guide
+attempts to act as a "check-list" when adding new resources.
+
+NOTE: This guide is not intended to be exhaustive, or even particularly
+detailed. For that, refer to the documentation which exists in the codebase --
+this document should act as a jumping-off point.
+
+=== **HTTP**
+* Add endpoints for either the internal or external API
+** xref:../nexus/src/external_api/http_entrypoints.rs[The External API] is customer-facing, and provides interfaces for both developers and operators
+** xref:../nexus/src/internal_api/http_entrypoints.rs[The Internal API] is internal, and provides interfaces for services on the Oxide rack (such as the Sled Agent) to call
+** Register endpoints in the `register_endpoints` method (https://github.com/oxidecomputer/omicron/blob/1dfe47c1b3122bc4f32a9c517cb31b1600581ea2/nexus/src/external_api/http_entrypoints.rs#L84[Example])
+** These endpoints typically call into the *Application* layer, and do not access the database directly
+* Inputs and Outputs
+** Input parameters are defined in https://github.com/oxidecomputer/omicron/blob/main/nexus/types/src/external_api/params.rs[params.rs] (https://github.com/oxidecomputer/omicron/blob/1dfe47c1b3122bc4f32a9c517cb31b1600581ea2/nexus/types/src/external_api/params.rs#L587-L601[Example])
+** Output views are defined in https://github.com/oxidecomputer/omicron/blob/main/nexus/types/src/external_api/views.rs[views.rs] (https://github.com/oxidecomputer/omicron/blob/1dfe47c1b3122bc4f32a9c517cb31b1600581ea2/nexus/types/src/external_api/views.rs#L270-L274[Example])
+
+=== **Lookup & Authorization**
+* Declare a new resource-to-be-looked-up via `lookup_resource!` in xref:../nexus/src/db/lookup.rs[lookup.rs] (https://github.com/oxidecomputer/omicron/blob/1dfe47c1b3122bc4f32a9c517cb31b1600581ea2/nexus/src/db/lookup.rs#L557-L564[Example])
```

Comment:
Notes for myself:

- It's kinda weird that the `lookup_resource` and `authz_resource` are separate; I *think* they both need to be used together to be functional
- Populating the list of resources in `oso_generic.rs` feels like it could be automated
- Generating the lookup methods on `LookupPath` feels like it could be automated
- I believe these structures actually depend on a `db::model` object existing (after all, how else would they know what to fetch?), which is another layer of coupling that's subtle here. I wonder if there's a way to consolidate all these tightly coupled pieces a little better, or to actually make them more loosely coupled.